lib/common/strings.c

```diff
@@ -359,8 +366,14 @@ crm_get_msec(const char *input)
         input++;
     }
 
-    // Reject negative and unparsable inputs
-    if ((scan_ll(input, &msec, -1, &units) == EINVAL) || (msec < 0)) {
+    rc = scan_ll(input, &msec, PCMK__PARSE_INT_DEFAULT, &units);
+
+    if ((rc == ERANGE) && (msec > 0)) {
+        crm_warn("'%s' will be clipped to %lld", input, msec);
+
+    } else if ((rc != pcmk_rc_ok) || (msec < 0)) {
+        crm_warn("'%s' is not a valid time duration: %s",
+                 input, ((rc == pcmk_rc_ok)? "Negative" : pcmk_rc_str(rc)));
```

Comment:
It makes sense to have the info about an invalid value. But OTOH, the parsing for the sbd env `SBD_DELAY_START` has been more or less expecting no warnings from crm_get_msec().

`SBD_DELAY_START` is such a special env which can be configured either to be a boolean value or a time duration. As long as the value is not crm_is_true(), it is passed to crm_get_msec():

https://github.com/ClusterLabs/sbd/blob/42455eb4298fa4eae03330ca1a1bc373ece58c8f/src/sbd-inquisitor.c#L991-L994

But it's now starting to get warnings with a valid boolean but false value:

```
warning: crm_get_msec: 'no' is not a valid time duration: Invalid argument
```

I'm not sure what could be the best way. If we shouldn't downgrade  the warning here to be "info", then sbd would probably have to as well first check if a value is an appropriate boolean false for instance with crm_str_to_boolean().

Reply:
I think sbd should check for boolean first. Pacemaker does something similar for `in_ccm` in `unpack_node_member()`. @wenningerk